### PR TITLE
Helm chart: Add support for custom bucket policies

### PR DESCRIPTION
**Pull request by bdomars**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license] (https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
Adds a way to set custom policies via `mc anonymous set-json` for buckets created via the Helm chart.

## Motivation and Context
We have a lot of custom policies set via the UI on buckets and If we try to use the Helm chart to create these buckets it will always override the set policies.

## How to test this PR?
Uncomment example `bucket3` in `values.yaml` and verify that a custom policy is set using `mc anonymous get-json myminio/bucket3`.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: helm/minio/values.yaml [hunk 1]
```diff
@@ -422,6 +422,42 @@ buckets: []
...
+  #                             "10.42.0.0/16",
+  #                         ]
+  #                     }
+  #                 }
+  #             }
+  #         ]
+  #     }
+  #   purge: false
+  #   versioning: true
+  #   objectlocking: false
```
Comment (harshavardhana) [inline helm/minio/values.yaml]: this is already referenced at line #424 - we do not need double references.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: helm/minio/values.yaml (hunk 1 above)
Comment (bdomars) [reply helm/minio/values.yaml]: Indeed, removed the extra attributes on bucket3 to only demonstrate the custom policy attributes.

**Review by jiuker — APPROVED** [review]

File: helm/minio/values.yaml [hunk 2]
```diff
@@ -422,6 +422,39 @@ buckets: []
...
+  #                     "s3:ListBucket"
+  #                 ],
+  #                 "Resource": [
+  #                     "arn:aws:s3:::bucket3",
+  #                     "arn:aws:s3:::bucket3/*"
+  #                 ],
+  #                 "Condition": {
+  #                     "IpAddress": {
+  #                         "aws:SourceIp": [
+  #                             "10.42.0.0/16",
```
Comment (shtripat) [inline helm/minio/values.yaml]: Remove the last `,` at the end in this sample JSON.

**Review by shtripat — APPROVED** [review]
LGTM. Small correction in values.yaml sample JSON.

**Comment by bdomars** [comment]
So is this okay to merge now?

**Comment by LinAnt** [comment]
Any reason not to merge this? :)

**Comment by harshavardhana** [comment]
Will consolidate this and merge them.

**Pull request closed without merging**